src/main.h

```diff
@@ -0,0 +1,77 @@
+#ifndef MAIN_H
```

Comment:
Why is this fine Main.h / why do we have main.cpp

why not renderlib/something more descriptive, considering it is not actually a main / the main is `Main.hs`